modules/platform/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/service/PluggableAuthenticationService.java

```diff
@@ -62,200 +63,109 @@
-            specificAuthChains.put(desc.name, desc);
-        } else if (extensionPoint.equals(EP_LOGINSCREEN)) {
-            LoginScreenConfig newConfig = (LoginScreenConfig) contribution;
-            registerLoginScreenConfig(newConfig);
-        }
+        });
     }
 
     @Override
-    public void unregisterContribution(Object contribution, String extensionPoint, ComponentInstance contributor) {
-
-        if (extensionPoint.equals(EP_AUTHENTICATOR)) {
-            AuthenticationPluginDescriptor descriptor = (AuthenticationPluginDescriptor) contribution;
-            authenticatorsDescriptors.remove(descriptor.getName());
-            log.debug("unregistered AuthenticationPlugin: " + descriptor.getName());
-        } else if (extensionPoint.equals(EP_LOGINSCREEN)) {
-            LoginScreenConfig newConfig = (LoginScreenConfig) contribution;
-            unregisterLoginScreenConfig(newConfig);
-        }
-    }
-
-    private void mergeDescriptors(AuthenticationPluginDescriptor newContrib) {
-        AuthenticationPluginDescriptor oldDescriptor = authenticatorsDescriptors.get(newContrib.getName());
-
-        // Enable/Disable
-        oldDescriptor.setEnabled(newContrib.getEnabled());
-
-        // Merge parameters
-        Map<String, String> oldParameters = oldDescriptor.getParameters();
-        oldParameters.putAll(newContrib.getParameters());
-        oldDescriptor.setParameters(oldParameters);
-
-        oldDescriptor.setStateful(newContrib.getStateful());
-
-        if (newContrib.getClassName() != null) {
-            oldDescriptor.setClassName(newContrib.getClassName());
-        }
-
-        oldDescriptor.setNeedStartingURLSaving(newContrib.getNeedStartingURLSaving());
+    public void stop(ComponentContext context) throws InterruptedException {
+        authenticators = null;
+        authChain = null;
+        authPluginChain = null;
+        sessionManagers = null;
     }
 
     // Service API
 
     public List<String> getStartURLPatterns() {
-        return startupURLs;
+        return this.<StartURLPatternDescriptor> getRegistryContribution(EP_STARTURL)
+                   .map(StartURLPatternDescriptor::getStartURLPatterns)
+                   .orElse(Collections.emptyList());
     }
 
     public List<String> getAuthChain() {
-        return authChain;
+        return Collections.unmodifiableList(authChain);
     }
 
     public List<String> getAuthChain(HttpServletRequest request) {
-
-        if (specificAuthChains == null || specificAuthChains.isEmpty()) {
-            return authChain;
-        }
-
-        SpecificAuthChainDescriptor desc = getAuthChainDescriptor(request);
-
-        if (desc != null) {
-            return desc.computeResultingChain(authChain);
-        } else {
+        List<SpecificAuthChainDescriptor> specificAuthChains = getRegistryContributions(EP_SPECIFIC_CHAINS);
+        if (specificAuthChains.isEmpty()) {
```

Comment:
This class was full of checks to make pseudo optimizations so that's kind of a remnant of this, but yes: if there are no specific chains we return the "standard" chain, otherwise we compute specific chains for this request.
Said differently: if there are no contributions, we do not bother computing the specific chain for this request.

I could rewrite it as is if you think it's better:
```
        String specificAuthChainName = getSpecificAuthChainName(request);
        if (specificAuthChainName == null) {
            return authChain;
        }
        return this.<SpecificAuthChainDescriptor> getRegistryContribution(EP_SPECIFIC_CHAINS, specificAuthChainName)
                   .map(desc -> desc.computeResultingChain(authChain))
                   .orElse(authChain);
```
instead of:
```
        List<SpecificAuthChainDescriptor> specificAuthChains = getRegistryContributions(EP_SPECIFIC_CHAINS);
        if (specificAuthChains.isEmpty()) {
            return authChain;
        }
        return getAuthChainDescriptor(request).map(desc -> desc.computeResultingChain(authChain)).orElse(authChain);
```

Reply:
Got it, maybe just a comment about the optimizations is enough.